kong-ngx-build

```diff
@@ -387,6 +407,83 @@ main() {
   succ "Build finished in $SECONDS seconds. Enjoy!"
 }
 
+parse_version() {
+  [[ -z $1 ]] && fatal 'missing arg $1 when invoking parse_version()'
+  [[ -z $2 ]] && fatal 'missing arg $2 when invoking parse_version()'
+
+  local subj=$1
+  local -n ver=$2
+  local re='^(-rc[0-9]+$)?[^0-9]*([0-9a-zA-Z]*)(.*?)$'
+
+  while [[ $subj =~ $re ]]; do
+    if [[ ${BASH_REMATCH[1]} != "" ]]; then
+      ver+=( "${BASH_REMATCH[1]}" )
+    fi
+
+    if [[ ${BASH_REMATCH[2]} != "" ]]; then
+      ver+=( "${BASH_REMATCH[2]}" )
+    fi
+
+    subj="${BASH_REMATCH[3]}"
+    if [[ $subj == "" ]]; then
+      break
+    fi
+  done
+}
+
+parse_nginx_core_version() {
+  [[ -z $1 ]] && fatal 'missing arg $1 when invoking parse_nginx_core_version()'
+
+  local nginx_ver
+
+  parse_version $1 nginx_ver
+
+  echo "${nginx_ver[0]}.${nginx_ver[1]}.${nginx_ver[2]}"
+}
+
+version_eq() {
+  local version_a version_b
+
+  parse_version $1 version_a
+  parse_version $2 version_b
+
+  # Note that we are indexing on the b components, ie: 1.11.100 == 1.11
+  for index in "${!version_b[@]}"; do
+    [[ "${version_a[index]}" != "${version_b[index]}" ]] && return 1
+  done
+
+  return 0
+}
+
+version_lt() {
+  local version_a version_b
+
+  parse_version $1 version_a
+  parse_version $2 version_b
+
+  for index in "${!version_a[@]}"; do
+    [[ "${version_a[index]}" < "${version_b[index]}" ]] && return 0
+    [[ "${version_a[index]}" > "${version_b[index]}" ]] && return 1
```

Comment:
These comparisons do not work on all cases, since these are string comparisons `"99" > "100"`

Reply:
Ok, let's add more test cases covering those then, thanks for the catch :+1: